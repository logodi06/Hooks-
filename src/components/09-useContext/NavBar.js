import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
  return (
    // <nav>
    //     <ul>
    //         <li>
    //             <Link to="/">Home</Link>
    //         </li>
    //         <li>
    //             <Link to="./about">About</Link>       
    //         </li>
    //         <li> 
    //             <Link to="./login">Login</Link>
    //         </li>
    //     </ul>
    // </nav>
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">
            <Link to="/" className="navbar-brand">useContext</Link>
           
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    {/* <Link to="/" className="nav-link" aria-current="page" >Home</Link>
                    <Link to="/about" className="nav-link" >About</Link>
                    <Link to="/login" className="nav-link" >Login</Link> */}

                    {/* El link y NavLink son iguales y hacen lo mismo, solo que
                    el Navlink permite identificar que componente se est√° seleccionando
                    y esta activa el "active" automaticamente */}
                    <NavLink  to="/" className="nav-link" aria-current="page" >Home</NavLink>
                    <NavLink  to="/about" className="nav-link" >About</NavLink>
                    <NavLink  to="/login" className="nav-link" >Login</NavLink>
                </div>
             </div>
        </div>
    </nav>
    );
};
