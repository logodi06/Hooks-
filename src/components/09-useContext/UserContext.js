import {createContext} from 'react';

//Va a exportar un componente, por eso el nombre tiene sintaxis de componente

//Esto crea un contex, todo lo que se defina aquí se podrá usar en 
//todos sus hijos
export const UserContext = createContext(null);