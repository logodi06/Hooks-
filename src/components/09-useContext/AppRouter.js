import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route,
    Navigate
} from "react-router-dom";
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
  return (
    <Router>
        <div>
            {/* Se crea una lista con las páginas a las que se puede navegar como menú */}
            <NavBar />
            <div className='container '>
            {/* Definición de rutas */}
                <Routes>
                    {/* Con el exact le decimos que la ruta debe ser examente como se degine */}
                    <Route exact path="/" element = {<HomeScreen />} />
                    <Route exact path="/about" element = {<AboutScreen />} />
                    <Route exact path="/login" element = {<LoginScreen />} />
                    <Route exact path="/login" element = {<LoginScreen />} />
                    {/* Ruta para que se redireccione cuando se coloque un path que no este definido */}
                    <Route path="*" element={<Navigate to ="/" />}/>
                

                </Routes>
            </div>
        </div>
    </Router>
  )
};
