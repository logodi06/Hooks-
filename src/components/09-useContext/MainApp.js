import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';


export const MainApp = () => {

   const [user, setUser] = useState({});

  return(
    //   Todo lo que se defina en el </UserContext> va a estar definido para todos sus hijos
    // Sus hijos es todo aquello que este dentro de el
    <UserContext.Provider value={{
        // user: user,
        // setUser: setUser
        //Esto es lo mismo que arriba pero de la forma corta
        user,
        setUser
    }}>
        <AppRouter />
    </UserContext.Provider>
  )
};
