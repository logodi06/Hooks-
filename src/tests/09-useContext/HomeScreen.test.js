import { HomeScreen } from "../../components/09-useContext/HomeScreen";
import {mount} from 'enzyme';
import { UserContext } from "../../components/09-useContext/UserContext";

describe('Pruebas en <HomeScreen />', () => { 
    
    const user = {
        name: 'Lorena',
        email: 'lore@gmail.com'
    } 
    
    //el shallow solo renderiza el primer componente que es el useContext
    //en este caso necesitamos el homeScreen por lo cual se usa mount
    const wrapper = mount(
        //Para evitar que nos muestre error porque al principio tiene el useContext
        <UserContext.Provider value={{
            user        
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('Debe de mostrar el componente <HomeScreen />', () => { 
        expect(wrapper).toMatchSnapshot();        
     })
 })