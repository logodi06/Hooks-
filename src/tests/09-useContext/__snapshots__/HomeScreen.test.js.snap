// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> Debe de mostrar el componente <HomeScreen /> 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Lorena",
   "email": "lore@gmail.com"
}
  </pre>
</div>
`;
